resources/x509_cert_gen/gen_keystore.sh

```diff
@@ -59,27 +59,38 @@ arr_hosts=( $hosts )
 echo Loop through nodes
 for host in "${arr_hosts[@]}"
 do
-	ip=`echo $host|sed 's/\"\([^:]*\):.*/\1/'`
-	# Copy private key and chain file to a node:/opt/couchbase/var/lib/couchbase/inbox
-	echo "Setup Certificate for ${ip}"
-	${SSH} root@${ip} "mkdir ${INBOX}" 2>/dev/null || true
-	${SCP} chain.pem root@${ip}:${INBOX}
-	${SCP} pkey.key root@${ip}:${INBOX}
-	${SSH} root@${ip} "chmod o+rx ${INBOX}${CHAIN}"
-	${SSH} root@${ip} "chmod o+rx ${INBOX}${NODE}.key"
-
-	# Upload ROOT CA and activate it
-	curl -s -o /dev/null --data-binary "@./${ROOT_CA}.pem" \
-    	http://${ADMINCRED}@${ip}:8091/controller/uploadClusterCA
-	curl -sX POST http://${ADMINCRED}@${ip}:8091/node/controller/reloadCertificate
-
-	# Enable client cert
-	if ${USE_JSON} ;then
-	    POST_DATA='{"state": "enable","prefixes": [{"path": "subject.cn","prefix": "","delimiter": ""}]}'
-            curl -s -H "Content-Type: application/json" -X POST -d "${POST_DATA}" http://${ADMINCRED}@${ip}:8091/settings/clientCertAuth
-	else
-            curl -s -d "state=enable" -d "delimiter=" -d "path=subject.cn" -d "prefix=" http://${ADMINCRED}@${ip}:8091/settings/clientCertAuth
-        fi
+  # Copy private key and chain file to a node:/opt/couchbase/var/lib/couchbase/inbox
+  echo "Setup Certificate for ${ip}"
+
+  if  [[ ${host:1:1} == "[" ]]
+  then
+      ip=`echo $host|sed "s/.*\[//;s/\].*//;"`
+      ip="[${ip}]"
+      new_ip=`echo $ip|sed "s/.*\[//;s/\].*//;"`
+      ${SSH} root@${new_ip} "mkdir ${INBOX}" 2>/dev/null || true
+      ${SCP} chain.pem root@${ip}:${INBOX}
+      ${SCP} pkey.key root@${ip}:${INBOX}
+      ${SSH} root@${new_ip} "chmod 777 ${INBOX}${CHAIN}"
```

Comment:
chmod 777 is not recomended. Is it required for ipv6 VMs?

Reply:
yes it is required .